app/guid-file/route.ts

```diff
@@ -54,11 +62,92 @@ export default class GuidFile extends Route {
+            JSON.stringify(this.structuredData) : JSON.stringify({ isAccessibleForFree : true });
+
+        const scriptTagsData = {
+            type: 'application/ld+json',
+            src: `osf.io/${parentGuidString}/metadata/?format=google-dataset-json-ld`,
+            content: jsonString,
+        };
+
+        const metaTags: HeadTagDef[] = this.metaTags.getHeadTags(metaTagsData);
+        const scriptTags: HeadTagDef[] = this.scriptTags.getHeadTags(scriptTagsData);
+        const allTags: HeadTagDef[] = metaTags.concat(scriptTags);
+
+        // Concatenate meta and script tags if both, only meta tags or none
+        if (metaTags && scriptTags) {
+            this.set('headTags', allTags);
+        } else if (metaTags) {
+            this.set('headTags', metaTags);
+        }
+        // Rebuild head tags and clear blocker
         this.headTagsService.collectHeadTags();
         blocker.done();
     }
 
+    async returnStructuredData(parentGuid: string,  environment: string): Promise<any> {
+        const url = `/${parentGuid}/metadata/?format=google-dataset-json-ld`;
+        let jsonLD: object = {};
+
+        if (environment === 'production') {
+            assert(`Currently on ${environment}`, Boolean('production'));
+        } else if (environment === 'staging' || environment === 'staging2' || environment === 'staging3') {
+            if (environment === 'staging') {
+                assert(`Currently on ${environment}`, Boolean('staging'));
+            } else if (environment === 'staging2') {
+                assert(`Currently on ${environment}`, Boolean('staging2'));
+            } else if (environment === 'staging3') {
+                assert(`Currently on ${environment}`, Boolean('staging3'));
+            }
+        } else if (environment === 'development') {
+            assert(`Currently on ${environment}`, Boolean('development'));
+        } else if (environment === 'test') {
+            assert(`Currently on ${environment}`, Boolean('test'));
+        } else {
+            throw new Error('Environment not found.');
+        }
+
+        // Override url
+        config.OSF.url.replace(/\/$/, url);
+
+        let jsonFetch : object | void;
+        try {
+            jsonFetch = await this.returnJSON(url);
+
+            if (jsonFetch && (typeof(jsonFetch) === 'object')) {
+                jsonLD = jsonFetch;
+                const JSONLDString = JSON.stringify(jsonLD);
+                this.set('structuredData', JSONLDString);
+            }
+            return jsonLD;
+        } catch (e) {
+            throw new Error(this.intl.t('general.structured_data.json_ld_retrieval_error'));
+        }
+    }
+
+    async returnJSON(url: string) {
+        const ajax = await this.currentUser.authenticatedAJAX({
+            method: 'GET',
+            url: getHref(url),
+            headers: {
+                'Content-Type': 'application/json',
+            },
+        });
+        return ajax;
+    }
+
```

Comment:
These each individually required awaits to resolve the Promises. If there is a better way to format this, we can have a pair programming session. Previously when I had it encapsulated all in one method, it would not properly resolve in Ember as only one await could be used in a method at a time as a modifier. Placing the calls in separate methods, similar to functional programming, where the return value is then passed as a resolved variable seemed to work. I can test this further though and we can discuss results and potential alternatives in the meeting.

Reply:
Also, these methods are now in the script-tags service.

Reply:
You many need to `await` multiple times, but I do not believe you should need to declare multiple functions to await. This is adding redundant function calls to the call stack and is not functional programming, which relies heavily on passing functions around to other higher-order functions. Remove most of the `try`, `if` and `catch` and see if you can reduce it to 2 `await`s  without adding new additional functions.

Reply:
Ok, I have simplified the strucutre. We will still need the try/catch block as it was breaking a large number of tests for the overview page without it.